src/records/RecMutex.cc

```diff
@@ -45,7 +45,7 @@ rec_mutex_acquire(RecMutex *m)
 {
   ink_thread this_thread = ink_thread_self();
 
-  if (m->thread_holding != this_thread) {
+  if (!pthread_equal(m->thread_holding, this_thread)) {
```

Comment:
Using `pthread_equal` is recommended to compare `pthread_t` (`ink_thread` is an alias of `pthread_t`).